src/main/java/fi/aalto/cs/apluscourses/intellij/actions/OpenSubmissionNotificationAction.java

```diff
@@ -0,0 +1,48 @@
+package fi.aalto.cs.apluscourses.intellij.actions;
+
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationAction;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import fi.aalto.cs.apluscourses.intellij.notifications.Notifier;
+import fi.aalto.cs.apluscourses.intellij.notifications.SubmissionRenderingErrorNotification;
+import fi.aalto.cs.apluscourses.model.SubmissionResult;
+import fi.aalto.cs.apluscourses.model.UrlRenderer;
+import org.jetbrains.annotations.NotNull;
+
+public class OpenSubmissionNotificationAction extends NotificationAction {
+
+  @NotNull
+  private final SubmissionResult submissionResult;
+
+  @NotNull
+  private final UrlRenderer submissionRenderer;
```

Comment:
why have a renderer here, if the message is being passed via `Notification Bus`?

Reply:
now I see... sorry :clap:

Reply:
No problem I would've asked the same thing :smile: